### PR TITLE
[WIP] First draft of prmrun support

### PR DESCRIPTION
**Pull request by jctanner**
# DO NOT MERGE

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
become method pmrun

##### ANSIBLE VERSION

```
2.3
```

##### SUMMARY

Adds support for a pmrun privilege-escalation/become method based on documented examples.

### CODE REVIEW

**Comment by bcoca** [comment]
does pmrun not allow you to chose which user you are going to run the command as?

**Comment by jctanner** [comment]
@bcoca not that i've seen so far.

EDIT: 

It can according to this man page: http://documents.software.dell.com/privilege-manager-for-unix/6.0/administrators-guide/privilege-manager-programs/pmrun

**Comment by jctanner** [comment]
```
<10.0.0.107> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=testuser -o ConnectTimeout=10 -o ControlPath=/home/jtanner/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.0.0.107 '/bin/sh -c '"'"'pmrun su -c /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-wugtyclmbwqdrarvpafrnsqxdkjcyfxb; /usr/bin/python /home/testuser/.ansible/tmp/ansible-tmp-1481926397.47-203780706409843/command.py; rm -rf "/home/testuser/.ansible/tmp/ansible-tmp-1481926397.47-203780706409843/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
```

**Comment by bcoca** [comment]
manpage uses -u but your  output is chaining su
```pmrun su -c /bin/sh -c ```

File: lib/ansible/playbook/play_context.py [hunk 1]
```diff
@@ -569,6 +569,12 @@ def detect_ksu_prompt(b_data):
                 else:
                     becomecmd = '%s -u %s %s' % (exe, self.become_user, command)
 
+            elif self.become_method == 'pmrun':
+
+                exe = self.become_exe or 'pmrun'
+                prompt = 'Enter UPM user password:'
+                becomecmd = '%s %s' % (exe, command)
```
Comment (bcoca) [inline lib/ansible/playbook/play_context.py]: according to manpage this should look like ```'%s -u %s %s' % (exe, self.become_user, command)```

File: lib/ansible/playbook/play_context.py [hunk 2]
```diff
@@ -569,6 +569,13 @@ def detect_ksu_prompt(b_data):
                 else:
                     becomecmd = '%s -u %s %s' % (exe, self.become_user, command)
 
+            elif self.become_method == 'pmrun':
+
+                exe = self.become_exe or 'pmrun'
+                exe += ' su -c'
```
Comment (bcoca) [inline lib/ansible/playbook/play_context.py]: why are you chaining `su` here?

File: lib/ansible/playbook/play_context.py [hunk 3]
```diff
@@ -569,6 +569,13 @@ def detect_ksu_prompt(b_data):
...
+                prompt = 'Enter UPM user password:'
+                becomecmd = '%s %s' % (exe, command)
```
Comment (bcoca) [inline lib/ansible/playbook/play_context.py]: from manpage we should support setting the become_user, like so:
```  becomecmd = '%s -u %s %s' % (exe, self.become_user, command)```

**Pull request closed without merging**